5. Linked List/mid/LinkedListMidPointArnab.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * LinkedListMidPointArnab.java
+ * Find mid point of a Linked List
+ * Description:-
+ * Initialize head node.Traverse the linked list to find the size() of the list.Find mid point of that list by dividing the size()
+ * by 2 then+1.Then traverse the Linked List to the mid point and then return the data of mid point.
+ * Time Complexity-O(n) Extra space complexity-O(n)
+ * @author [codebook-2000](https://github.com/codebook-2000)
+ */
+ 
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+
+class Node<T> {
+    T data;   //Its a generic class
+    Node next;//next stores the reference of the next node
+
+    public Node(T data) {
+        this.data = data;
+    }//initializes the data part of the object
+}
+
+class LinkedListMidPointArnab {
+    static int solve(Node head) { //This method calculates the mid point
```

Comment:
This is a very good question to hone linked lists. Your solution is good but in linked list don't try to solve by using i based iteration, instead use pointer based iteration. I am sure you can find a better solution.